### ISSUE TITLE
Stream reading error

### ISSUE THREAD

**Issue by SergeyGulik** [TrakHound/MTConnect.NET#38]
Hi Patrick,

In the changeset 9dbdbedf1d27ec27f6f6ea275c0cbd689be4c371 you introduced a problem:
![image](https://user-images.githubusercontent.com/10059586/213457167-e835c473-f8ba-4af4-a525-03d6d03c0320.png)
You ignore the result of ReadAsync and b never get -1 as before your change.
Instead it should be something like
![image](https://user-images.githubusercontent.com/10059586/213457452-17ddc311-055e-4454-b541-5989d0a7865d.png)

At least there is one more place above in the same file with the same problem.

Best regards,
Sergii Gulyk.

**Comment by SergeyGulik** [comment]
It went unnoticed because usually this stream is "endless" and you stop reading it by timeout. Except a HAAS machine that does not support an "interval" parameter in _sample_ query and clearly reports it. In such case this code hangs forever aggravating memory by endlessly adding \r to the _line_.

**Comment by PatrickRitchie** [comment]
@SergeyGulik Thanks for finding this and yes I believe this would prevent the issue you described if I understand the issue correctly. It looks like I misunderstood how the stream.ReadAsync() method worked and it may explain a few instances where I saw strange behavior during testing.

I'm working on an update that is available in the new [dev](https://github.com/TrakHound/MTConnect.NET/tree/dev) branch which I included this change. Let me know if this issue is fixed with that update. 

*Note that I moved the MTConnect.Clients.Rest.MTConnectClient class to MTConnect.Clients.MTConnectHttpClient*
